src/flb_pack.c

```diff
@@ -923,13 +924,23 @@ flb_sds_t flb_pack_msgpack_to_json_format(const char *data, uint64_t bytes,
             case FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP:
             /* Format the time, use microsecond precision not nanoseconds */
                 gmtime_r(&tms.tm.tv_sec, &tm);
-                s = strftime(time_formatted, sizeof(time_formatted) - 1,
+                s = strftime(time_formatted, sizeof(time_formatted),
                              FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP_FMT, &tm);
 
+                max_len = sizeof(time_formatted) - s;
                 len = snprintf(time_formatted + s,
-                               sizeof(time_formatted) - 1 - s,
+                               max_len,
                                ".%06" PRIu64,
                                (uint64_t) tms.tm.tv_nsec / 1000);
+                if (len >= max_len) {
+                    if (json_format == FLB_PACK_JSON_FORMAT_LINES ||
```

Comment:
I think in reality what we care about at this point is not what `json_format` holds but to ensure that `out_buf` is not `NULL`, in fact there's an error related to that in a different point in the function so, let's keep it as simple as possible.

Reply:
I have completely removed the check because it is safe to call the cleanup function with NULL.